File: eternal_return/src/main/java/org/EternalReturn/ERAnimal/ERAJEntityLoader.java
package org.EternalReturn.ERAnimal;

import com.google.gson.Gson;

import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

/**
 * Gson으로 animal.json 내의 야생동물 정보를 파싱, load()로 반환한다.
 * 생성 시 JSON파일의 경로가 필요하다.
 * */
public class ERAJEntityLoader {

    record JsonRoot(
            List<JsonArea> location
    ) {}

    record JsonArea(
            String name,
            List<JsonAnimal> table
    ) {}

    record JsonAnimal(
            String name,
            double[] pos,
            double[] rot
    ) {}

    private JsonRoot root;

    public ERAJEntityLoader(String jsonPath){
        try {
            String json = Files.readString(Path.of(jsonPath));
            Gson gson = new Gson();
            this.root = gson.fromJson(json, JsonRoot.class);

        } catch (Exception e) {
            throw new RuntimeException("Failed to load animal.json", e);
        }

    }

    public List<AreaERAnimalInfo> load(){
        List<AreaERAnimalInfo> areaInfoMap = new ArrayList<>();

        for (ERAJEntityLoader.JsonArea area : root.location()) {

            List<ERAnimalInfo> animals = new ArrayList<>();

            for (ERAJEntityLoader.JsonAnimal a : area.table()) {
                animals.add(new ERAnimalInfo(
                        a.name(),
                        a.pos()[0],
                        a.pos()[1],
                        a.pos()[2],
                        (float)a.rot()[0],
                        (float)a.rot()[1],
                        null
                ));
            }

            areaInfoMap.add(new AreaERAnimalInfo(area.name(), animals));
        }

        return areaInfoMap;
    }


    public static void main(String[] args){
        ERAJEntityLoader loader = new ERAJEntityLoader("animal.json");
        List<AreaERAnimalInfo> list = loader.load();

        for(AreaERAnimalInfo info : list){
            System.out.println(info.name());
            System.out.println(info);
        }

    }


}

File: eternal_return/src/main/java/org/eternalreturn/eranimal/ERAJEntityLoader.kt
package org.eternalreturn.eranimal

import com.google.gson.Gson
import java.nio.file.Files
import java.nio.file.Path

/**
 * Gson으로 animal.json 내의 야생동물 정보를 파싱, load()로 반환한다.
 * 생성 시 JSON파일의 경로가 필요하다.
 */
class ERAJEntityLoader(jsonPath: String) {
    @JvmRecord
    internal data class JsonRoot(
        val location: MutableList<JsonArea>?
    )

    @JvmRecord
    internal data class JsonArea(
        val name: String,
        val table: MutableList<JsonAnimal>?
    )

    @JvmRecord
    internal data class JsonAnimal(
        val name: String,
        val pos: DoubleArray,
        val rot: DoubleArray
    )

    private var root: JsonRoot

    init {
        try {
            val json = Files.readString(Path.of(jsonPath))
            val gson = Gson()
            this.root = gson.fromJson<JsonRoot>(json, JsonRoot::class.java)
        } catch (e: Exception) {
            throw RuntimeException("Failed to load animal.json", e)
        }
    }

    fun load(): MutableList<AreaERAnimalInfo> {
        val areaInfoMap: MutableList<AreaERAnimalInfo> = ArrayList<AreaERAnimalInfo>()

        for (area in root.location!!) {
            val animals: MutableList<ERAnimalInfo> = ArrayList<ERAnimalInfo>()

            for (a in area.table!!) {
                animals.add(
                    ERAnimalInfo(
                        a.name,
                        a.pos[0],
                        a.pos[1],
                        a.pos[2],
                        a.rot[0].toFloat(),
                        a.rot[1].toFloat()
                    )
                )
            }

            areaInfoMap.add(AreaERAnimalInfo(area.name, animals))
        }

        return areaInfoMap
    }

}
